pkg/network/usm/sharedlibraries/ebpf.go

```diff
@@ -336,19 +375,21 @@ func (e *EbpfProgram) Subscribe(callback LibraryCallback, libsets ...Libset) (fu
 		return nil, err
 	}
 
-	e.callbacksMutex.Lock()
-	defer e.callbacksMutex.Unlock()
-
+	var unsubscribers []func()
 	for _, libset := range libsets {
-		e.getLibsetData(libset).callbacks[&callback] = struct{}{}
+		// e.libsets is only modified when creating the EbpfProgram struct,
+		// which is a singleton with a mutex in the GetEBPFProgram function. As
+		// Golang maps are thread-safe for reads, we don't need here a mutex to
+		// access it. subscribe() will the libset-specific callback, to avoid
+		// locking on all callbacks for all libsets.
+		unsub := e.libsets[libset].subscribe(callback)
```

Comment:
we might have a libset that does not exists in `e.libsets`

Reply:
The function `IsLibsetValid` ensures that the libset is present in the `LibsetToLibSuffixes` map, which is the one we use to fill the `e.libsets` map. If someone adds a new libset and forgets to add it to `LibsetToLibSuffixes` it will be caught and an error will be launched in `InitWithLibsets`.